### PR TITLE
enhancement #555: improved endpoint for displaying teachers in a specific discipline

### LINKED ISSUE #555: Update endpoint that returns all disciplines of the group
To create an event, we need to improve the endpoint to return the teachers of a particular discipline:

image

You should implement the possibility of searching for a teacher by a certain type of discipline. For this, you can add query.

### PR DESCRIPTION
**Pull request by SharpDevOps10**
enhancement #555: improved endpoint for displaying teachers in a specific discipline

- added disciplineType for getting teachers
- added swagger documentation
- fixed bug with rating
- added responses

closes #555

### CODE REVIEW

File: src/v2/api/controllers/DisciplineController.ts [hunk 1]
```diff
@@ -31,11 +41,33 @@ export class DisciplineController {
...
+    name: 'disciplineType',
+    enum: DisciplineTypeEnum,
+    required: false,
+  })
+  @ApiOkResponse({
+    type: DisciplineTeachersResponse,
+  })
+  @ApiBadRequestResponse({
+    description: `\n
+                  InvalidQueryException:
```
Comment (f-ogoog) [inline src/v2/api/controllers/DisciplineController.ts:55]: needless spaces

File: src/v2/api/controllers/DisciplineController.ts [hunk 2]
```diff
@@ -31,11 +41,33 @@ export class DisciplineController {
...
+                    Type of discipline must be a field of enum
+                  InvalidEntityIdException:
+                    Discipline with such id is not found`,
+  })
+  @ApiForbiddenResponse({
+    description: `\n
+                  NoPermissionException:
```
Comment (f-ogoog) [inline src/v2/api/controllers/DisciplineController.ts:62]: needless spaces

**Review by f-ogoog — APPROVED** [review]

**Comment by JessFreak** [comment]
image
Undocumented exception

**Comment by JessFreak** [comment]
image
image

Undocumented / Unhandled exception

**Review by JessFreak — CHANGES_REQUESTED** [review]
image
Undocumented exception

**Review by JessFreak — CHANGES_REQUESTED** [review]
image
image
Undocumented / Unhandled exception

**Review by JessFreak — APPROVED** [review]

**Review by JessFreak — APPROVED** [review]

File: src/v2/api/services/DisciplineService.ts [hunk 3]
```diff
@@ -19,11 +21,17 @@ export class DisciplineService {
...
   }
 
-  async getTeachers (id: string) {
+  async getTeachers (disciplineId: string, disciplineType: DisciplineTypeEnum) {
+    const role = TeacherRoleAdapter[disciplineType];
     const disciplineTeachers = await this.disciplineTeacherRepository.findMany({
       where: {
+        roles: {
+          some: {
+            role: role,
```
Comment (just-hoshinon) [inline src/v2/api/services/DisciplineService.ts:30]: can be simplified to just `role`

**Review by just-hoshinon — CHANGES_REQUESTED** [review]

**Review by just-hoshinon — APPROVED** [review]

**Pull request merged**